js/index.js

```diff
@@ -1,20 +1,73 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
 
+  // console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span');
+  // console.log("We are printing price :", price);
+  const quantity = product.querySelector('.quantity input');
+  // console.log("We are printing quantity :", quantity);
+
+  let priceValue = price.innerText;
+  // console.log("Print price", priceValue);
+  let quantityValue = quantity.value;
+  // console.log("Print quantity", quantityValue);
+
+  let subtotal = priceValue * quantityValue;
+  // console.log("Subtotal", subtotal);
+
+  product.querySelector('.subtotal span').innerText = subtotal;
+
+  return subtotal;
+
+  
   //... your code goes here
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  // const singleProduct = document.querySelector('.product');
+  // updateSubtotal(singleProduct);
   // end of test
 
   // ITERATION 2
   //... your code goes here
+ 
+ 
+  const product = document.getElementsByClassName("product");
+  // console.log("here is what prints product:", product[0])
+  // console.log("here is subtotal for 1st product", updateSubtotal(product[0]))
+  // console.log("here is subtotal for 2nd product", updateSubtotal(product[1]))
+  let total = 0;
+  ///////////////////////////
+  for (let i = 0; i <product.length; i++){
+    // console.log("Here it is each subtotal", updateSubtotal(product[i]));
+    total += updateSubtotal(product[i]);
+    // console.log("the total is =", total);
+  }
+  /////////////////////////// WHY NOT WORKING???
+  // product.forEach(oneproduct => {
+  //   total += updateSubtotal(oneproduct);
+  // });
```

Comment:
In this case, forEach cannot be used since it is a method only applied to arrays, and product is an object!